doc/operator-notes.md

```diff
@@ -62,3 +62,40 @@ If tooling is being used to generate Ignition configs, the tooling _should_ gene
 
 [selinux]: https://selinuxproject.org/page/Main_Page
 [restorecon]: https://linux.die.net/man/8/restorecon
+
+## Partition Reuse Semantics
+
+The `wipePartitionEntry` and `shouldExist` flags control what Ignition will do when it encounters an existing partition. `wipePartitionEntry` specifies whether Ignition is permitted to delete partition entries in the partition table.  `shouldExist` specifies whether a partition with that number should exist or not (it is invalid to specify a partition should not exist and specify its attributes, such as `size` or `label`).
+
+The following table shows the possible combinations of `shouldExist`, `wipePartitionEntry`, whether or not a partition with the specified number is present, and the action Ignition will take
+
+| partition present | shouldExist | wipePartitionEntry | action Ignition takes
+| ----------------- | ----------- | ------------------ | ---------------------
+| false             | false       | false              | Do nothing
+| false             | false       | true               | Do nothing
+| false             | true        | false              | Create specified partition
+| false             | true        | true               | Create specified partition
+| true              | false       | false              | Fail
+| true              | false       | true               | Delete existing partition
+| true              | true        | false              | Check if existing partition matches the specified one, fail if it does not
+| true              | true        | true               | Check if existing partition matches the specified one, delete existing partition and create specified partition if it does not match
+
+### Partition Matching
+A partition matches if all of the specified attributes (`label`, `start`, `size`, `uuid`, and `typeGuid`) are the same. Specifying `uuid` or `typeGuid` as an empty string is the same as not specifying them. When 0 is specified for start or size, Ignition checks if the existing partition's start / size match what they would be if all of the partitions specified were to be deleted (if allowed by wipPartitionEntry), then recreated if `shouldExist` is true.
+
+### Partition number 0
+Specifying `number` as 0 will use the next available partition number. Partition number 0 is disallowed on disks with partitions that specify `shouldExist` as 0. If `number` is not specified it will be treated as 0.
```

Comment:
I'm having problems following this. How can a partition specify `shouldExist` as 0? `shouldExist` is a boolean.